templates/defineOperation.dot

```diff
@@ -1,13 +1,19 @@
 {-| {{= it.doc }}
-{{? it.input && it.input.members }}
-__Parameters__
+{{? it.requiredParams }}
+__Required Parameters__
 
-{{~ it.input.members :m }}* `{{= m.key }}` __:__ `{{= m.value.type }}`
+{{~ it.requiredParams :p }}* `{{= p.key }}` __:__ `{{= p.value.type }}`
 {{~}}
 {{?}}
 -}
-{{= it.name }} : {{? it.input }}{{~ it.input.members :m }}{{= m.value.type }} -> {{~}}{{?}}AWS.ServiceConfig -> AWS.Http.UnsignedRequest {{= it.output.type }}
-{{= it.name }} {{? it.input }}{{~ it.input.members :m }}{{= m.key }} {{~}}{{?}}serviceConfig =
+{{= it.name }} :
+    {{~ it.requiredParams :p }}{{= p.value.type }}
+    -> {{~}}{{? it.optionalParams.length }}({{= it.optionsName }} -> {{= it.optionsName }})
+    -> {{?}}AWS.Http.UnsignedRequest {{= it.output.type }}
+{{= it.name }} {{~ it.requiredParams :p }}{{= p.key }} {{~}}{{? it.optionalParams.length }}setOptions {{?}}={{? it.optionalParams.length }}
+  let
+    options = setOptions ({{= it.optionsName }} {{= it.optionalParams.map(() => 'Nothing').join(' ') }})
```

Comment:
Good idea. Do you know how to apply a function to a list of arguments? I can't find `apply`.

I tried doing something like this:

```elm
        options =
            setOptions
                ((List.repeat 3 Nothing)
                    |> List.foldl (\a cur -> cur a) ListCertificatesOptions
                )
```

but the compiler just hung, didn't report an error, but didn't complete either. I guess because the signature of the reducer was invalid